### PR TITLE
Only create target directory if it doesn’t exist.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Fixes https://github.com/Homebrew/homebrew-cask/issues/58564.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Was just about to submit a PR for this.

Won’t it work if we just change line 43 from `if target.dirname.parent.writable?` to `if target.dirname.writable?` (i.e. remove `parent.`)? The `dirname` of `target` is the directory we’re moving it to, so why check its parent?

I’ve tried `if target.dirname.writable?` both with `app` and with `java`, and they both worked.

**Comment by reitermarkus** [comment]
I have updated it with a different solution.

File: Library/Homebrew/cask/artifact/moved.rb [hunk 1]
```diff
@@ -40,13 +40,13 @@ def move(force: false, command: nil, **options)
         end
 
         ohai "Moving #{self.class.english_name} '#{source.basename}' to '#{target}'."
-        if target.dirname.parent.writable?
+        if target.dirname.ascend.detect(&:directory?).writable?
```
Comment (vitorgalvao) [inline Library/Homebrew/cask/artifact/moved.rb]: `style` suggests `find` over `detect`.

**Comment by robamaton** [comment]
Awesome, thanks! Any idea when this might be released?

**Pull request merged**